engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -799,6 +807,10 @@ public void monitorHeartbeat(boolean waitForRecovery)
     processEvents();
 
     committedWindowId = updateCheckpoints(waitForRecovery);
+    if (lastCommittedWindowId != committedWindowId) {
```

Comment:
No, the lastCommittedWindowId is not restored it will be set to -1. The updateCheckpoints will return the committed windowId based on current checkpoints and then this variable is set inside the in condition.